### PR TITLE
python312Packages.beniget: 0.4.1 -> 0.4.2.post1

### PR DESCRIPTION
**Pull request by dotlambda**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/pythran/default.nix [hunk 1]
```diff
@@ -51,9 +57,9 @@ buildPythonPackage rec {
     ln -s '${lib.getDev xsimd}'/include/xsimd pythran/
   '';
 
-  nativeBuildInputs = [ setuptools ];
+  build-system = [ setuptools ];
 
-  propagatedBuildInputs = [
```
Comment (doronbehar) [inline pkgs/development/python-modules/pythran/default.nix:56]: Thanks for these changes, I'd like semantic changes to appear in a separate commit.

File: pkgs/development/python-modules/pythran/default.nix [hunk 2]
```diff
@@ -76,6 +82,7 @@ buildPythonPackage rec {
   disabled = !isPy3k;
 
   meta = {
+    changelog = "https://github.com/serge-sans-paille/pythran/blob/${src.rev}/Changelog";
```
Comment (doronbehar) [inline pkgs/development/python-modules/pythran/default.nix:85]: Another semantic change.

**Review by doronbehar — COMMENTED** [review]
Thanks for the PR! Looks pretty good to me.

File: pkgs/development/python-modules/beniget/default.nix [hunk 3]
```diff
@@ -1,25 +1,36 @@
...
+
+  pythonImportsCheck = [ "beniget" ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
 
   meta = {
     description = "Extract semantic information about static Python code";
     homepage = "https://github.com/serge-sans-paille/beniget";
     license = lib.licenses.bsd3;
+    maintainers = [ ];
```
[the remark this answers is not in the document]
Comment (dotlambda) [reply pkgs/development/python-modules/beniget/default.nix:34]: No thanks

**Comment by doronbehar** [comment]
I'm sorry but my perfectionism wouldn't let me not split the modernization changes to separate commits.

**Comment by dotlambda** [comment]
No worries at all.

**Pull request merged**